packages/hooks/use-real-shape/src/index.ts

```diff
@@ -8,15 +8,15 @@ export function useRealShape<T extends HTMLElement>(ref: RefObject<T | null>) {
     width: 0,
     height: 0,
   });
-  const updateShape = () => {
+  const updateShape = useCallback(() => {
     if (!ref?.current) return;
 
     const {width, height} = getRealShape(ref.current);
 
     setState({width, height});
-  };
+  }, [ref]);
```

Comment:
i think we don't need to add `ref` to dependency array here.

Reply:
Yes, it looks like what you said from the code. It's just that my local ESLint rule "react-hooks/exhaustive-deps" thinks that "ref" needs to be one of the dependencies, and adding it doesn't have any impact, so I kept it.

Reply:
let's remove it to keep it consistent with other code and align with the project lint.